include/grpc/event_engine/event_engine.h

```diff
@@ -255,6 +255,47 @@ class EventEngine : public std::enable_shared_from_this<EventEngine> {
     /// values are expected to remain valid for the life of the Endpoint.
     virtual const ResolvedAddress& GetPeerAddress() const = 0;
     virtual const ResolvedAddress& GetLocalAddress() const = 0;
+
+    /// A method which allows users to query whether an Endpoint implementation
+    /// supports a specified extension. The name of the extension is provided
+    /// as an input.
+    ///
+    /// An extension could be any type with a unique string id. Each extension
+    /// may support additional capabilities and if the Endpoint implementation
+    /// supports the queried extension, it should return a valid pointer to the
+    /// extension type.
+    ///
+    /// E.g., use case of an EventEngine::Endpoint supporting a custom
+    /// extension.
+    ///
+    /// class CustomEndpointExtension {
+    ///  public:
+    ///.   static constexpr std::string Name() {
```

Comment:
`///.` should just be `///`

And I still recommend a `static constexpr std::string`, I'm not sure what benefit there is in this being a function.

Reply:
I think static constexpr constraints to only values known at compile time. By retaining it as a function, we can return values evaluated at run-time as well.

Reply:
Updated it now in the documentation to use a static constexpr std::string.